server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/TabletBase.java

```diff
@@ -233,20 +233,19 @@ public Tablet.LookupResult lookup(List<Range> ranges, List<KVEntry> results,
   Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, ScanParameters scanParams)
       throws IOException {
 
-    try {
-      while (context.getLowMemoryDetector().isRunningLowOnMemory(context, DetectionScope.SCAN,
-          () -> {
-            return isUserTable;
-          }, () -> {
-            log.info("Not starting next batch because low on memory, extent: {}", extent);
-            server.getScanMetrics().incrementScanPausedForLowMemory();
-            Thread.sleep(500);
-          })) {}
-    } catch (InterruptedException e) {
-      // Reset the interrupt state on this thread
-      Thread.interrupted();
-      throw new IOException("Pause for low memory interrupted", e);
-    }
+    while (context.getLowMemoryDetector().isRunningLowOnMemory(context, DetectionScope.SCAN, () -> {
+      return isUserTable;
+    }, () -> {
+      log.info("Not starting next batch because low on memory, extent: {}", extent);
+      server.getScanMetrics().incrementScanPausedForLowMemory();
+      try {
+        Thread.sleep(500);
+      } catch (InterruptedException e) {
+        Thread.interrupted();
```

Comment:
Should this be `Thread.currentThread().interrupt()` ?

Reply:
Addressed in 679612b